src/TestData.hs

```diff
@@ -28,6 +40,34 @@ data Account ts = Account
 deriving instance AllHave Show (Account ts) => Show (Account ts)
 deriving instance AllHave Eq (Account ts)   => Eq (Account ts)
 
+jonathan :: User 'Data
+jonathan = User { userId = (Id "1"), userName = ( Name "Jonathan"), userBestFriend = sandy, userFriends = [] }
+
+sandy :: User 'Data
+sandy = User { userId = (Id "2"), userName = ( Name "Sandy"), userBestFriend = jonathan, userFriends = []}
+
+getUserResolver :: (Arg "id" Id) -> User 'Query -> IO (User 'Response)
+getUserResolver a q
+    | (getArg a) == (Id "1") = pure $ hydrate jonathan q
+    | (getArg a) == (Id "2") = pure $ hydrate sandy q
+    | otherwise = pure $ hydrate jonathan q
+
+getAllUsersResolver :: User 'Query -> IO ([User 'Response])
+getAllUsersResolver q = pure $ (flip hydrate q) <$> [sandy, jonathan]
```

Comment:
Is hydrate smart enough that this could be just `pure $ hydrate [sandy, jonathan] q`? If not let's file a bug so it can be.